Sphinx timeouts

On GitLab CI the Sphinx build sometimes fail like this:

```
Running Sphinx v1.6.7
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 39 source files that are out of date
updating environment: 39 added, 0 changed, 0 removed
reading sources... [  2%] addendum/canonical-structures
reading sources... [  5%] addendum/extended-pattern-matching
reading sources... [  7%] addendum/extraction
reading sources... [ 10%] addendum/generalized-rewriting
reading sources... [ 12%] addendum/implicit-coercions
/usr/lib/python3/dist-packages/sphinx/environment/__init__.py:704: RemovedInSphinx17Warning: html_use_smartypants option is deprecated. Smart quotes are on by default; if you want to disable them, use the smartquotes option.
  RemovedInSphinx17Warning)

Exception occurred:
  File "/usr/lib/python3.6/subprocess.py", line 410, in run
    stderr=stderr)
subprocess.TimeoutExpired: Command '['/builds/coq/coq/bin/coqdoc', '--body-only', '--no-glob', '--no-index', '--no-externals', '-s', '--html', '--stdout', '--utf8', '/tmp/coqdoc-bw4qgij5.v']' timed out after 2 seconds
The full traceback has been saved in /tmp/sphinx-err-ipjepkix.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile.doc:57: recipe for target 'sphinx' failed
```

(from https://gitlab.com/coq/coq/-/jobs/68511985)

For comparison, this is a successful run on the same commit: https://gitlab.com/coq/coq/-/jobs/68541041.

@cpitclaudel or @maximedenes Would you mind having a look?

--- a/doc/tools/coqrst/coqdoc/main.py
+++ b/doc/tools/coqrst/coqdoc/main.py
@@ -40,7 +40,7 @@ def coqdoc(coq_code, coqdoc_bin=None):
         os.write(fd, COQDOC_HEADER.encode("utf-8"))
         os.write(fd, coq_code.encode("utf-8"))
         os.close(fd)
-        return check_output([coqdoc_bin] + COQDOC_OPTIONS + [filename], timeout = 2).decode("utf-8")
+        return check_output([coqdoc_bin] + COQDOC_OPTIONS + [filename], timeout = 10).decode("utf-8")
     finally:
         os.remove(filename)
 

